tasks/components/OrganizationForm.tsx

```diff
@@ -55,103 +54,110 @@ const defaultOrganizationFormTranslations: Record<Languages, OrganizationFormTra
   }
 }
 
-type OrganizationDTO = {
+type OrganizationGeneralInfoDTO = {
   shortName: string,
   longName: string,
   email: string,
   isVerified: boolean
 }
 
+// TODO make sure the Organization type only has the used values shortName, longName, email, isVerified
 export type OrganizationFormProps = {
-  organization?: OrganizationDTO,
-  onSave?: (organization: OrganizationDTO) => void
+  organization?: OrganizationGeneralInfoDTO,
+  onChange: (organization: OrganizationGeneralInfoDTO, isValid: boolean) => void,
+  isShowingErrorsDirectly?: boolean
 }
 
 export const OrganizationForm = ({
   language,
   organization = { shortName: '', longName: '', email: '', isVerified: false },
-  onSave = () => undefined
+  onChange = () => undefined,
+  isShowingErrorsDirectly = false
 }: PropsWithLanguage<OrganizationFormTranslation, OrganizationFormProps>) => {
   const translation = useTranslation(language, defaultOrganizationFormTranslations)
-  const [newOrganization, setNewOrganization] = useState(organization)
-  const [touched, setTouched] = useState({ shortName: false, longName: false, email: false })
+  const [touched, setTouched] = useState({ shortName: isShowingErrorsDirectly, longName: isShowingErrorsDirectly, email: isShowingErrorsDirectly })
+
   const minShortNameLength = 2
-  const minLongNameLength = 3
+  const minLongNameLength = 4
   const maxShortNameLength = 16
   const maxLongNameLength = 64
+  const maxMailLength = 320
 
   const inputErrorClasses = tw('border-hw-negative-500 focus:border-hw-negative-500 focus:ring-hw-negative-500 border-2')
   const errorClasses = tw('text-hw-negative-500 text-sm')
   const inputClasses = tw('mt-1 block rounded-md w-full border-gray-300 shadow-sm focus:outline-none focus:border-hw-primary-500 focus:ring-hw-primary-500')
 
-  let shortNameErrorMessage: string | undefined
-  if (newOrganization.shortName === '') {
-    shortNameErrorMessage = translation.required
-  } else if (newOrganization.shortName.length < minShortNameLength) {
-    shortNameErrorMessage = translation.tooShort(minShortNameLength)
-  } else if (newOrganization.shortName.length > maxShortNameLength) {
-    shortNameErrorMessage = translation.tooLong(maxShortNameLength)
+  function validateShortName(organization: OrganizationGeneralInfoDTO): string | undefined {
+    if (organization.shortName === '') {
+      return translation.required
+    } else if (organization.shortName.length < minShortNameLength) {
+      return translation.tooShort(minShortNameLength)
+    } else if (organization.shortName.length > maxShortNameLength) {
+      return translation.tooLong(maxShortNameLength)
+    }
+  }
+
+  function validateLongName(organization: OrganizationGeneralInfoDTO): string | undefined {
```

Comment:
TypeScripts type system is clever enough to infer the types, try to reduce the amount of manual return types.